### PR TITLE
Coalesce rows

### PR DESCRIPTION
**Pull request by tims**
For #88 

/wip

### CODE REVIEW

**Comment by tims** [comment]
@pradithya @woop @budi This is the PR I mentioned to you in chat. Take a look if you're interested in the implementation details. It's still needs a few changes.

eg
* change the import spec options field to be sourceOptions, but I want to wait for another PR so we have less merge conflicts.
* group by entityname +entitykey, instead of FeatureRowKey, but this is dependent on removing history from the serving stores first.

File: ingestion/src/main/java/feast/ingestion/transform/CoalesceFeatureRows.java [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+import org.apache.beam.sdk.values.PCollection;
+import org.apache.beam.sdk.values.PCollection.IsBounded;
+import org.apache.beam.sdk.values.TypeDescriptors;
+import org.joda.time.Duration;
+import org.joda.time.Instant;
+
+public class CoalesceFeatureRows extends
+    PTransform<PCollection<FeatureRow>, PCollection<FeatureRow>> {
+
+  private static final Comparator<Timestamp> TIMESTAMP_COMPARATOR = Comparator
```
Comment (pradithya) [inline ingestion/src/main/java/feast/ingestion/transform/CoalesceFeatureRows.java]: we can use `Timestamps.compare(t1, t2)` for this

File: ingestion/src/main/java/feast/ingestion/ImportJob.java [hunk 2]
```diff
@@ -170,6 +175,12 @@ public void expand() {
                     ParDo.of(new RoundEventTimestampsDoFn())),
             pFeatureRows.getErrors());
 
+    if (jobOptions.isCoalesceRowsEnabled()) {
+      pFeatureRows = pFeatureRows.apply("foo", new CoalescePFeatureRows(
```
Comment (pradithya) [inline ingestion/src/main/java/feast/ingestion/ImportJob.java]: "foo" can be replaced with "Coalesce Feature Row"

File: ingestion/src/main/java/feast/ingestion/options/JobOptions.java [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+  private boolean coalesceRowsEnabled;
+  private long coalesceRowsDelaySeconds;
+  private long coalesceRowsTimeoutSeconds;
+
+  @JsonProperty(value = "sample.limit")
+  public long getSampleLimit() {
+    return sampleLimit;
+  }
+
+  public void setSampleimit(long sampleLimit) {
```
Comment (syvlabs) [inline ingestion/src/main/java/feast/ingestion/options/JobOptions.java]: typo on method name

**Comment by tims** [comment]
I've included not writing to the feature stores as requested.

**Comment by tims** [comment]
Related to #87

**Comment by tims** [comment]
should this be on or off by default?

**Comment by pradithya** [comment]
Is this in-scope for 0.1.0?

**Comment by zhilingc** [comment]
@tims done, I think we're good to merge it in

**Comment by zhilingc** [comment]
sorry, sausage fingers :(

**Comment by tims** [comment]
boo... @pradithya can you lgtm this?

**Pull request merged**